server/s3_motr_writer.cc

```diff
@@ -372,6 +372,48 @@ void S3MotrWiter::create_object_failed() {
   s3_log(S3_LOG_DEBUG, "", "Exiting\n");
 }
 
+void S3MotrWiter::sync_content(std::function<void(void)> on_success,
+                               std::function<void(void)> on_failed) {
+  s3_log(S3_LOG_INFO, request_id, "Entering\n");
+  handler_on_success = std::move(on_success);
+  handler_on_failed = std::move(on_failed);
+
+  struct m0_op *sync_op = NULL;
+  int rc = m0_sync_op_init(&sync_op);
+  if (rc != 0) {
+    m0_op_fini(sync_op);
+    m0_op_free(sync_op);
+  }
```

Comment:
If sync_op_init() fails it will run sync_entity_add(), can you change it.
Do the same after sync_entity_add().